python-sdk/tests/databases/test_snowflake.py

```diff
@@ -165,3 +165,58 @@ def test_snowflake_load_options_empty():
     assert not load_options.empty()
     load_options.file_options = {}
     assert load_options.empty()
+
+
+def test_storage_integrations_params():
+    """
+    test that if `storage_integration` is not passed to `load_file_to_table_natively` method either via
+    `native_support_kwargs` or via `SnowflakeLoadOptions` it raises an exception.
+    """
+    table = Table(conn_id="fake-conn")
+    file = File(path="azure://data/homes_main.ndjson")
+    database = SnowflakeDatabase(conn_id=table.conn_id, load_options=SnowflakeLoadOptions())
+    with pytest.raises(ValueError) as e:
+        database.load_file_to_table_natively(source_file=file, target_table=table)
+    assert e.match(
+        "Param 'storage_integration' is required. Please pass either in `native_support_kwargs` or"
+        " `SnowflakeLoadOptions.copy_options`"
+    )
+
+
+def test_storage_integrations_params_in_native_support_kwargs():
+    """
+    Test passing of storage_integrations in native_support_kwargs
+    """
+    table = Table(conn_id="fake-conn")
+    file = File(path="azure://data/homes_main.ndjson")
+    database = SnowflakeDatabase(conn_id=table.conn_id, load_options=SnowflakeLoadOptions())
+
+    with mock.patch("astro.databases.snowflake.SnowflakeDatabase.create_stage"), mock.patch(
+        "astro.databases.snowflake.SnowflakeDatabase._copy_into_table_from_stage"
+    ) as _copy_into_table_from_stage, mock.patch(
+        "astro.databases.snowflake.SnowflakeDatabase.evaluate_results"
+    ):
+        database.load_file_to_table_natively(
+            source_file=file,
+            target_table=table,
+            native_support_kwargs={"storage_integration": "some_integrations"},
+        )
+
+
+def test_storage_integrations_params_in_load_options():
+    """
+    Test passing of storage_integrations in SnowflakeLoadOptions
+    """
+    table = Table(conn_id="fake-conn")
+    file = File(path="azure://data/homes_main.ndjson")
+    database = SnowflakeDatabase(
+        conn_id=table.conn_id,
+        load_options=SnowflakeLoadOptions(copy_options={"storage_integration": "some_integrations"}),
+    )
+
+    with mock.patch("astro.databases.snowflake.SnowflakeDatabase.create_stage"), mock.patch(
+        "astro.databases.snowflake.SnowflakeDatabase._copy_into_table_from_stage"
+    ) as _copy_into_table_from_stage, mock.patch(
+        "astro.databases.snowflake.SnowflakeDatabase.evaluate_results"
+    ):
+        database.load_file_to_table_natively(source_file=file, target_table=table)
```

Comment:
What are we checking as part of this test?

Reply:
in the previous test `test_storage_integrations_params` we made sure that if the `storage_integration` is not passed the `load_file_to_table_natively` will raise the ValueError exception. Here we are making sure to pass `storage_integration` via `SnowflakeLoadOptions` don't.

Reply:
Sure